### ISSUE TITLE
Allow changing triple-click (and double-click) behaviour

### ISSUE THREAD

**Issue by DarrenCook** [Microsoft/vscode#57633]
Currently when I triple-click a word, the whole line gets selected.

I'm finding this annoying, as I seem to be doing it by accident, when intending to double-click to select just one word.

So I want to be able to customize triple-click to behave identically to double-click.

Searching user settings I cannot find anything for "triple". I also cannot find anything for controlling double-click behaviour. Or even how to define the action for "select word" and "select line". (I've also checked the keymap config, in case mouse actions were in there.)

(That is surprising, so if the config options I want already exist, then my feature request is to make them more easily discoverable!)

**Comment by Logerfo** [comment]
Also, the triple click behavior should be configurable between selecting the line break and not selecting the line break. Currently, it always select the line break (if there is one).

**Comment by HaraldKorneliussen** [comment]
From the Apple HIG: _Triple-clicking selects the next logical unit, as defined by the application. In a word-processing document, triple-clicking in a word selects the paragraph containing the word…. In some contexts—in a programming language, for example—it may be appropriate to allow users to select both the left and right parentheses (or braces or brackets) in a pair, as well as all the characters between them, by double-clicking either one of them.”_

I think those guidelines make a lot of sense, and what the hierarchy of logical units looks like will vary between e.g. a markdown file and a java file.

**Comment by noelfelix** [comment]
Yeah, customizable (n-)click event behavior would be fantastic.

Eg I would rather triple click expand selection, personally.

**object**.property -> **object.property** rather than the line.

**Comment by ykray** [comment]
Any updates on this? Would be an amazing productivity-customization feature.

**Comment by DrMiaow** [comment]
Can this be re-visited please?

**Comment by movy** [comment]
Upvoting! Expanding selection to brackets/braces/quotation marks via triple click would be amazing improvement to copy-paste routines.

**Comment by alecgargett** [comment]
Personally I would love it if double click stays the same, triple click selects everything up until the nearest space OR line-break OR bracket/parentheses (so that you can easily select a whole URL for example by triple clicking, while still being able to select just one part of a URL by double clicking) & then four clicks for the whole line, five for the whole paragraph, etc.

**Comment by bay-systems** [comment]
Upvote.  Double-click selects a word, determined by "editor.wordSeparators".  Subsequent clicks should expand the selection logically, not just to the entire line as it currently does.

**Comment by dcnetw0rk** [comment]
i need this too. it would be perfect if we can set a custom regex match or tell the cursor to highlight only characters and exclude leading/trailing whitespace on triple click -- iterm2 has this feature. 🤷‍♀️

**Comment by Lovecircle** [comment]
Upvote! Especially dangerous when I'm trying to double click a word and then delete it but instead end up deleting the contents of the whole file!

**Comment by roy-tate** [comment]
I just need a way to disable triple-click.  Even a registry entry!

**Comment by DarrenCook** [comment]
Yes: this is still driving me crazy, and having an option to simply disable triple-click would be more than sufficient.
(My original post over-complicated it, talking about customization. That can come later.)

**Comment by YevhenNJSN** [comment]
Yes, it would be great to customize n-click actions in VSCode. 'Expand selection' on triple-click seems like killer feature)

**Comment by droolrockworm** [comment]
Double click should select the nearest word, breaking on any kind delimiter including space. Next (triple) click should select up to nearest delimiter, this time ignoring (including) space. N clicks past that could just continue this pattern. I don't understand why this should even be customizable, just have it be the literal way it works forever. Again, is there a reason not to? It's so easy to just keep clicking up until you get the right block you want, even up to the paragraph or function level

**Comment by torrilhon** [comment]
cascading/hierarchical n-clicks, ideally customizable, PLEASE!

**Comment by tarsisexistence** [comment]
up:)

**Comment by tahaum** [comment]
Would be nice!
